testdata/p4tc_samples_outputs/simple_extern_example_control_blocks.c

```diff
@@ -0,0 +1,342 @@
+            ext_params->inst_id = 2;
+            ext_params->index = 2;
+
+            __builtin_memcpy(ext_val->out_params, &arg_val_0, sizeof(struct reg_val_t ));
+            bpf_p4tc_extern_md_write(skb, ext_params, sizeof(ext_params), ext_val, sizeof(ext_val));
+;
+            /* nh_table_0.apply() */
+            {
+                /* construct key */
+                struct p4tc_table_entry_act_bpf_params__local params = {
+                    .pipeid = p4tc_filter_fields.pipeid,
+                    .tblid = 1
+                };
+                struct ingress_nh_table_key key;
+                __builtin_memset(&key, 0, sizeof(key));
+                key.keysz = 32;
+                key.field0 = hdr->ipv4.srcAddr;
+                struct p4tc_table_entry_act_bpf *act_bpf;
+                /* value */
+                struct ingress_nh_table_value *value = NULL;
+                /* perform lookup */
+                act_bpf = bpf_p4tc_tbl_read(skb, &params, sizeof(params), &key, sizeof(key));
+                value = (struct ingress_nh_table_value *)act_bpf;
+                if (value == NULL) {
+                    /* miss; find default action */
+                    hit = 0;
+                } else {
+                    hit = value->hit;
+                }
+                if (value != NULL) {
+                    /* run action */
+                    switch (value->action) {
+                        case INGRESS_NH_TABLE_ACT_INGRESS_EXT_REG: 
+                            {
+                                __builtin_memset(ext_params, 0, sizeof(struct p4tc_ext_bpf_params));
+                                ext_params->pipe_id = p4tc_filter_fields.pipeid;
+                                ext_params->ext_id = 1;
+                                ext_params->inst_id = 1;
+                                ext_params->index = value->u.ingress_ext_reg.port_id;
+
+                                ext_val = bpf_p4tc_extern_md_read(skb, ext_params, sizeof(ext_params));
+                                if (!ext_val) 
+                                     return TC_ACT_SHOT;
+                                __builtin_memcpy(&val_0, ext_val->out_params, sizeof(u32 ));
+                                                                val_0 = (val_0 + 10);
+                                /* reg1_0.write(value->u.ingress_ext_reg.port_id, val_0) */
+                                __builtin_memset(ext_params, 0, sizeof(struct p4tc_ext_bpf_params));
+                                ext_params->pipe_id = p4tc_filter_fields.pipeid;
+                                ext_params->ext_id = 1;
+                                ext_params->inst_id = 1;
+                                ext_params->index = value->u.ingress_ext_reg.port_id;
+
+                                __builtin_memcpy(ext_val->out_params, &val_0, sizeof(u32 ));
+                                bpf_p4tc_extern_md_write(skb, ext_params, sizeof(ext_params), ext_val, sizeof(ext_val));
```

Comment:
I think the current implementation is correct. Changing to above will throw compiler error.
 error: passing 'struct p4tc_ext_bpf_params' to parameter of incompatible type 'struct p4tc_ext_bpf_params *'; remove *

Reply:
Sorry, now I see the typo.
`ext_params` and `ext_val` remain the same, however the sizes that should be passed are the sizes of the structs, not the pointers.

So it would be:

```
bpf_p4tc_extern_md_write(skb, *ext_params, sizeof(*ext_params), *ext_val, sizeof(*ext_val));
```